Sources/AEPOptimize/Optimize.swift

```diff
@@ -278,13 +287,17 @@ public class Optimize: NSObject, Extension {
         guard
             event.isPersonalizationDecisionResponse,
             let requestEventId = event.requestEventId,
-            updateRequestEventIdsInProgress.contains(where: { $0.key == requestEventId })
+            updateRequestEventIdsInProgress.shallowCopy.contains(where: { $0.key == requestEventId })
         else {
             Log.debug(label: OptimizeConstants.LOG_TAG,
                       """
                       Ignoring Edge event, either handle type is not personalization:decisions, or the response isn't intended for this extension.
                       """)
-            propositionsInProgress.removeAll()
+            // Removing all the values from thread-safe dictionary
+            // TODO: Introduce removeAll method to ThreadSafeDictionary in AEPCore
+            for key in propositionsInProgress.keys {
+                _ = propositionsInProgress.removeValue(forKey: key)
```

Comment:
Is there an edge case where another thread is adding another entry while we are removing the all in here?